src/api/entities/Portfolio/index.ts

```diff
@@ -212,6 +216,91 @@ export abstract class Portfolio extends Entity<UniqueIdentifiers, HumanReadable>
     return values(assetBalances);
   }
 
+  /**
+   * Retrieve the NFTs held in this portfolio
+   *
+   *  @param args.assets - array of NftCollection (or tickers) for which to fetch holdings (optional, all holdings are retrieved if not passed)
+   */
+  public async getNftsHeld(args?: {
+    assets: (string | NftCollection)[];
+  }): Promise<PortfolioNftHolding[]> {
+    const {
+      owner: { did },
+      _id: portfolioId,
+      context: {
+        polymeshApi: {
+          query: { portfolio },
+        },
+      },
+      context,
+    } = this;
+
+    const rawPortfolioId = portfolioIdToMeshPortfolioId({ did, number: portfolioId }, context);
+    const [exists, freeBalanceEntries, lockedBalanceEntries] = await Promise.all([
+      this.exists(),
+      portfolio.portfolioNFT.entries(rawPortfolioId),
+      portfolio.portfolioLockedNFT.entries(rawPortfolioId),
+    ]);
+
+    if (!exists) {
+      throw new PolymeshError({
+        code: ErrorCode.DataUnavailable,
+        message: notExistsMessage,
+      });
+    }
+
+    const queriedCollections = args?.assets.map(asset => asTicker(asset));
+    const seenTickers = new Set<string>();
+
+    const processBalance = (
+      entry: (typeof freeBalanceEntries)[0],
+      balanceRecord: Record<string, Nft[]>
+    ): void => {
+      const [key] = entry;
+
+      const ticker = tickerToString(key.args[1][0]);
+      const heldId = u64ToBigNumber(key.args[1][1]);
+
+      // ignore assets not specified if the user provided a filter arg
+      if (queriedCollections && !queriedCollections.includes(ticker)) {
+        return;
+      }
+
+      seenTickers.add(ticker);
+
+      if (!balanceRecord[ticker]) {
+        const nft = new Nft({ id: heldId, ticker }, context);
+
+        balanceRecord[ticker] = [nft];
+      } else {
+        const nft = new Nft({ id: heldId, ticker }, context);
+        balanceRecord[ticker].push(nft);
+      }
+    };
+
+    const freeBalances: Record<string, Nft[]> = {};
+    freeBalanceEntries.forEach(entry => processBalance(entry, freeBalances));
+
+    const lockedBalances: Record<string, Nft[]> = {};
+    lockedBalanceEntries.forEach(entry => processBalance(entry, lockedBalances));
+
+    const holdings: PortfolioNftHolding[] = [];
+    seenTickers.forEach(ticker => {
+      const free = freeBalances[ticker] || [];
+      const locked = lockedBalances[ticker] || [];
+      const total = new BigNumber(free.length + locked.length);
+
+      holdings.push({
+        asset: new NftCollection({ ticker }, context),
```

Comment:
```suggestion
        nftCollection: new NftCollection({ ticker }, context),
```
can we be explicit? OK with what it is right now as well

Reply:
I was thinking it would be easier to keep the interface similar to the fungible ones.

They are separate storage on chain and don't think they will ever be mixed, but if had a method that returned both mixed it can be hard to distinguish in TS without having an explicit `type` field.

I think I am going to go with just `collection`